core/conversion/evaluators/eval_util.cpp

```diff
@@ -118,6 +118,34 @@ void checkSequenceSize(int64_t n, int64_t dim, int64_t seq_size) {
   }
 }
 
+at::Tensor scalar_to_tensor_util(const at::Scalar& s, const at::Device device = at::kCPU) {
```

Comment:
This needs to be done in a safe manner since we are messing with types. We should at least throw an warning saying we are truncating the type. Ideally we could just optionally turn it on, but we don't have ConversionCtx available

Reply:
maybe we can also use the truncate_long_and_double in compile_settings to turn it on and throw an warning when this is enable, however, this means that we must have access to compile_settings from this function in the prim.cpp.

Reply:
I was considering adding ConversionCtx for evaluators. Maybe we just do that